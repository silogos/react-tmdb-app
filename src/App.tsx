import "./index.css";
import { BrowserRouter, Route, Routes } from "react-router";

import MovieListLayout from "@/layouts/MovieList";

import HomePage from "@/pages/Home";
import MovieDetailPage from "@/pages/MovieDetail";
import NowPlayingPage from "@/pages/NowPlaying";
import PopularPage from "@/pages/Popular";
import TopRatedPage from "@/pages/TopRated";
import UpComingPage from "@/pages/Upcoming";
import SearchPage from "@/pages/Search";

const App = () => {
  return (
    <>
      <div
        id="backdrop-overlay"
        className="fixed inset-0 bg-black bg-opacity-100 transition-opacity duration-1000"
      />
      <BrowserRouter>
        <Routes>
          <Route element={<MovieListLayout />}>
            <Route index element={<HomePage />} />
            <Route path="movies">
              <Route path="now-playing" element={<NowPlayingPage />} />
              <Route path="popular" element={<PopularPage />} />
              <Route path="top-rated" element={<TopRatedPage />} />
              <Route path="upcoming" element={<UpComingPage />} />
            </Route>
          </Route>
          <Route path="/movies/search" element={<SearchPage />} />
          <Route path="movie/:movieId" element={<MovieDetailPage />} />
        </Routes>
      </BrowserRouter>
    </>
  );
};

export default App;
