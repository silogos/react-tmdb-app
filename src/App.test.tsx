import { describe, expect, it } from "vitest";
import { render } from "@testing-library/react";
import App from "App";

describe("App", () => {
  it("should mount correctly", () => {
    const component = render(<App />);
    expect(component).toMatchSnapshot();
  });
});
