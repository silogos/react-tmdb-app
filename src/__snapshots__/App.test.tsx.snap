// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > should mount correctly 1`] = `
<body>
  <div>
    <div
      class="fixed inset-0 bg-black bg-opacity-100 transition-opacity duration-1000"
      id="backdrop-overlay"
    />
    <div
      class="relative w-full min-h-screen"
    >
      <section
        class="overflow-hidden py-4"
        id="hero"
      >
        <div
          class="mx-auto max-w-screen-xl px-4 py-14 lg:px-8 lg:py-20"
        >
          <h1
            class="text-center text-gray-300 text-4xl lg:text-7xl mb-4"
          >
            ðŸŽ¬ Movie App
          </h1>
          <p
            class="text-center text-gray-300 text-base lg:text-2xl"
          >
            (The Movie Database)
          </p>
        </div>
      </section>
      <section
        class="overflow-hidden py-4"
        id="movie-list"
      >
        <div
          class="mx-auto max-w-screen-xl px-6 lg:px-8"
        >
          <div
            class="flex flex-col lg:flex-row lg:items-center justify-between mb-4 gap-4"
          >
            <div
              class="flex flex-row flex-nowrap overflow-x-auto gap-4"
            >
              <a
                aria-current="page"
                class="px-4 py-2  text-nowrap rounded-lg bg-gray-50"
                data-discover="true"
                href="/"
              >
                All
              </a>
              <a
                class="px-4 py-2  text-nowrap rounded-lg bg-gray-400"
                data-discover="true"
                href="/movies/now-playing"
              >
                Now Playing
              </a>
              <a
                class="px-4 py-2  text-nowrap rounded-lg bg-gray-400"
                data-discover="true"
                href="/movies/popular"
              >
                Popular
              </a>
              <a
                class="px-4 py-2  text-nowrap rounded-lg bg-gray-400"
                data-discover="true"
                href="/movies/top-rated"
              >
                Top Rated
              </a>
              <a
                class="px-4 py-2  text-nowrap rounded-lg bg-gray-400"
                data-discover="true"
                href="/movies/upcoming"
              >
                Upcoming
              </a>
            </div>
            <form>
              <input
                class="w-full border-2 rounded-lg  px-4 py-2 bg-white"
                data-testid="search-input"
                name="query"
                placeholder="Search..."
                required=""
                type="search"
              />
            </form>
          </div>
          <div
            class="grid grid-cols-2 md:grid-cols-4 lg:grid-cols-5 gap-y-8 gap-4"
          >
            <div
              class="animate-pulse aspect-[8/12] rounded-lg bg-gray-700 bg-opacity-90"
            />
            <div
              class="animate-pulse aspect-[8/12] rounded-lg bg-gray-700 bg-opacity-90"
            />
            <div
              class="animate-pulse aspect-[8/12] rounded-lg bg-gray-700 bg-opacity-90"
            />
            <div
              class="animate-pulse aspect-[8/12] rounded-lg bg-gray-700 bg-opacity-90"
            />
            <div
              class="animate-pulse aspect-[8/12] rounded-lg bg-gray-700 bg-opacity-90"
            />
            <div
              class="animate-pulse aspect-[8/12] rounded-lg bg-gray-700 bg-opacity-90"
            />
            <div
              class="animate-pulse aspect-[8/12] rounded-lg bg-gray-700 bg-opacity-90"
            />
            <div
              class="animate-pulse aspect-[8/12] rounded-lg bg-gray-700 bg-opacity-90"
            />
            <div
              class="animate-pulse aspect-[8/12] rounded-lg bg-gray-700 bg-opacity-90"
            />
            <div
              class="animate-pulse aspect-[8/12] rounded-lg bg-gray-700 bg-opacity-90"
            />
            <div
              class="animate-pulse aspect-[8/12] rounded-lg bg-gray-700 bg-opacity-90"
            />
            <div
              class="animate-pulse aspect-[8/12] rounded-lg bg-gray-700 bg-opacity-90"
            />
            <div
              class="animate-pulse aspect-[8/12] rounded-lg bg-gray-700 bg-opacity-90"
            />
            <div
              class="animate-pulse aspect-[8/12] rounded-lg bg-gray-700 bg-opacity-90"
            />
            <div
              class="animate-pulse aspect-[8/12] rounded-lg bg-gray-700 bg-opacity-90"
            />
            <div
              class="animate-pulse aspect-[8/12] rounded-lg bg-gray-700 bg-opacity-90"
            />
            <div
              class="animate-pulse aspect-[8/12] rounded-lg bg-gray-700 bg-opacity-90"
            />
            <div
              class="animate-pulse aspect-[8/12] rounded-lg bg-gray-700 bg-opacity-90"
            />
            <div
              class="animate-pulse aspect-[8/12] rounded-lg bg-gray-700 bg-opacity-90"
            />
            <div
              class="animate-pulse aspect-[8/12] rounded-lg bg-gray-700 bg-opacity-90"
            />
          </div>
        </div>
      </section>
    </div>
  </div>
</body>
`;
